src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureSelector.cpp

```diff
@@ -45,24 +50,254 @@ namespace OpenMS
 
   MRMFeatureSelector::~MRMFeatureSelector() {}
 
-  void MRMFeatureSelector::optimize_Tr() {}
+  Int MRMFeatureSelector::_addVariable(LPWrapper& problem, String& name, bool bounded=true, double obj=1.) {
+    Int index = problem.addColumn();
+    if (bounded) {
+      problem.setColumnBounds(index, 0, 1, LPWrapper::DOUBLE_BOUNDED);
+    } else {
+      problem.setColumnBounds(index, 0, 1, LPWrapper::UNBOUNDED);
+    }
+    problem.setColumnName(index, name);
+    problem.setColumnType(index, LPWrapper::CONTINUOUS);
+    problem.setObjective(index, obj);
+    return index;
+  }
+
+  void MRMFeatureSelector::_addConstraint(LPWrapper& problem, size_t size, Int *indices_array, double *values_array, String name, double lb, double ub, LPWrapper::Type param) {
+    std::vector<Int> indices(indices_array, indices_array + size);
+    std::vector<double> values(values_array, values_array + size);
+    problem.addRow(indices, values, name, lb, ub, param);
+  }
+
+  std::vector<String> MRMFeatureSelectorScore::optimize(
+      std::vector<std::pair<double, String>> time_to_name, 
+      std::map< String, std::vector<Feature> > feature_name_map
+  ) {
+    std::cout << "=======START OPTIMIZE SCORE========" << std::endl;
+    std::unordered_set<std::string> variables;
+    LPWrapper problem;
+    problem.setObjectiveSense(LPWrapper::MIN);
+    for (size_t cnt1=0; cnt1 < time_to_name.size(); ++cnt1) {
```

Comment:
It seems possible to use the range-for here. It helps forgetting about things like `time_to_name[cnt1]`, which are easy to do mistakes with.